lib/virtmanager.pm

```diff
@@ -0,0 +1,632 @@
+package virtmanager;
+use testapi;
+
+our @ISA = qw(Exporter);
+our @EXPORT = qw(launch_virtmanager launch_xterm connection_details create_vnet create_new_pool create_new_volume clean_up_desktop create_netinterface sudo_root delete_netinterface create_guest);
+
+
+sub sudo_root {
+    type_string "sudo su -"; send_key "ret";
+    type_string get_var('ROOTPASSWORD'); send_key "ret";
+}
+
+sub launch_xterm() {
+    if (check_var("DESKTOP", "icewm") ) {
+	send_key "ctrl-alt-t", 10;
+    } elsif (check_var("DESKTOP", "gnome")) {
+	send_key "alt-f2", 10;
+	type_string "xterm\n", 50;
+    } else { 
+	# assuming default is gnome
+	send_key "alt-f2", 10;
+	type_string "xterm\n", 50;
+    }
+}
+
+sub launch_virtmanager() {
+    clean_up_desktop();
+    # start a console
+    launch_xterm();
+    # launch virt-manager
+    type_string "virt-manager";
+    send_key "ret", 1;
+    # root password
+    type_string get_var('ROOTPASSWORD');
```

Comment:
you want to add an assert_screen here. otherwise you may type the root password somewhere you don't control